src/Workspaces/Core/Portable/Workspace/ProjectSystem/IProjectSystemDiagnosticSource.cs

```diff
@@ -2,18 +2,12 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
 
 // TODO: see if we can get rid of this interface by appropriately rewriting HostDiagnosticUpdateSource to live at the workspaces layer.
 internal interface IProjectSystemDiagnosticSource
 {
-    void ClearAllDiagnosticsForProject(ProjectId projectId);
-    void ClearAnalyzerReferenceDiagnostics(AnalyzerFileReference fileReference, string language, ProjectId projectId);
-    void ClearDiagnosticsForProject(ProjectId projectId, object key);
     DiagnosticData CreateAnalyzerLoadFailureDiagnostic(AnalyzerLoadFailureEventArgs e, string fullPath, ProjectId projectId, string language);
```

Comment:
Seems like this can be a utility helper method now and IProjectSystemDiagnosticSource interface can be completely removed.

Reply:
Yup. I can work on that next